gobblin-utility/src/main/java/gobblin/util/ProxiedFileSystemWrapper.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.PrivilegedExceptionAction;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.SequenceFile;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.token.Token;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Throwables;
+import com.google.common.io.Closer;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+
+
+/**
+ * A wrapper class for generating a file system as a proxy user.
+ */
+public class ProxiedFileSystemWrapper {
+  private static final Logger LOG = LoggerFactory.getLogger(ProxiedFileSystemWrapper.class);
+  private FileSystem proxiedFs;
+
+  // Two authentication types for Hadoop Security, through TOKEN or KEYTAB.
+  public enum AuthType {
+    TOKEN,
+    KEYTAB;
+  }
+
+  /**
+   * Constructor for ProxiedFileSystemWrapper. It uses the passed parameters to create an instance of a proxiedFs.
+   * @param proxyUserName
+   * @param logInUserName
+   * @param properties
+   * @param authType is either TOKEN or KEYTAB.
+   * @param authPath is the KEYTAB location if the authType is KEYTAB; otherwise, it is the token file.
+   * @param uri File system URI.
+   * @throws IOException
+   * @throws InterruptedException
+   * @throws URISyntaxException
+   */
+  public ProxiedFileSystemWrapper(String proxyUserName, String logInUserName, State properties, AuthType authType,
```

Comment:
Why not add all the logic in this method into getProxiedFilesystem. All you have to do is create a `final FileSystem proxiedFileSystem` object before the `doAs` block, and then you can set `proxiedFileSystem` in the `doAs` block. For example: http://stackoverflow.com/questions/9545076/access-variables-of-outer-class-in-java

I think having this logic in `getProxiedFileSystem` makes more sense than having this logic in the constructor